### PR TITLE
option to wait for user to press 'return' after command execution

### PR DESCRIPTION
**Pull request by pirafrank**
Added option `-w` or `--wait` to wait for the user to press return/enter key after executing the command. This allows the command output not to disappear immediately after exit and to stay readable on screen.

Example below.

custom-shell

### CODE REVIEW

**Comment by AnirudhG07** [comment]
Oh wow! I like it. I will review it in detail tomorrow.

**Comment by AnirudhG07** [comment]
Why do we need a `.editorconfig` file?

**Comment by pirafrank** [comment]
Well mostly because of indentation, to avoid future contributors to mix tabs and spaces in the file. Lua does not have a strict style requirement about using tabs or spaces, and since you chose tabs .editorconfig just ensures those are used.

**Comment by AnirudhG07** [comment]
I guess it's fine. We need not add it to the repo. I understand the use case, please remove it. rest LGTM.

**Comment by pirafrank** [comment]
done

File: README.md [hunk 1]
```diff
@@ -65,11 +65,13 @@ To change these options, you can give the following arguments to the plugin:
 
 Check the keybindings below to see how to set these options.
 
+You can also add `--wait` or `-w` to make it wait for the user to press return key after executing the command. This allows the command output not to disappear immediately after exit and to stay readable on screen. Note that it is up to the command you run to decide whether to wait for user input or not, so this option may or may not be needed.
+
```
Comment (AnirudhG07) [inline README.md:69]: One last thing, can you add your preview from PR here too? Or if you want make another, sure your wish. Add it after line 68. Thanks!!!

Comment (pirafrank) [reply README.md:69]: Makes sense. I have registered a new one. Any pref for where to upload the gif, `.assets` in repo root or some external service ?

Comment (AnirudhG07) [reply README.md:69]: sure .assets work.

Comment (pirafrank) [reply README.md:69]: Fair, done.

**Review by AnirudhG07 — APPROVED** [review]
Thanks!

**Comment by AnirudhG07** [comment]
Hi @pirafrank, I am unable to figure out why `--wait` is not working. By this I mean, I am able to use `-w` but not `--wait` in my commands.
Can you help me out on this?? Would really appreciate your help.

**Comment by pirafrank** [comment]
> Hi @pirafrank, I am unable to figure out why `--wait` is not working. By this I mean, I am able to use `-w` but not `--wait` in my commands. Can you help me out on this?? Would really appreciate your help.

Hi there, sure. I had a look at the code I wrote in this PR and found no reason why it should not work (conditions are in AND).

Then I tested the plugin without the wait options (I have set `run = "plugin custom-shell --args=auto"`) and still the plugin hangs in the status bar. I have updated to and tested the plugin with the above configuration on `0.4.2` an it doesn't work. Tested both on Debian and macOS. I guess there's something not working with yazi `0.4.x`. PR was developed on `0.3.x` and the plugin was working back then.

I need some spare time to dig deeper on this.

**Comment by AnirudhG07** [comment]
I think this is because of the "confirm" which was changed to "interactive", and "confirm" is deprecated. I am looking into fixing that also. So far only changed the job params.

**Comment by AnirudhG07** [comment]
Let's discuss this in the above mentioned issue! Thanks

**Pull request merged**